File: cryptography/brute_force.js
function* generate(alphabet, maxLength) {
	if (maxLength <= 0) return
	for (const c of alphabet) {
		yield c
	}
	for (const c of alphabet) {
		for (const next of generate(alphabet, maxLength - 1)) {
			yield c + next
		}
	}
}

for (const c of generate('ab', 2)) {
	console.log(c)
}

File: cryptography/brute_force.py
def generate(alphabet, max_len):
    if max_len <= 0:
        return
    for c in alphabet:
        yield c
    for c in alphabet:
        for next in generate(alphabet, max_len-1):
            yield c + next


for p in generate('abc', 3):
    print(p)
